zerver/tests/test_create_video_call.py

```diff
@@ -8,55 +8,104 @@ def setUp(self) -> None:
+
+    def test_create_video_request_query_error(self) -> None:
         class MockResponse:
             def __init__(self) -> None:
-                self.status_code = 401
+                self.status_code = 400
+
+        with mock.patch('requests.post', return_value=MockResponse()):
+            result = self.client_get("/json/calls/complete_zoom_user")
+            self.assert_json_error(result, "No code specified")
+
+    def test_create_video_credential_error(self) -> None:
+        class MockResponse:
+            def __init__(self) -> None:
+                self.status_code = 400
 
             def raise_for_status(self) -> None:
-                raise Exception("Invalid request!")
+                raise Exception()
 
         with mock.patch('requests.post', return_value=MockResponse()):
-            result = self.client_get("/json/calls/create")
-            self.assert_json_success(result)
-            result_dict = json.loads(result.content.decode('utf-8'))
+            result = self.client_get("/json/calls/complete_zoom_user?code=code")
+            self.assert_json_error(result, "Invalid Zoom credentials")
 
-            # TODO: Arguably this is the wrong result for errors, but
-            # in any case we should test it.
-            self.assertEqual(result_dict['zoom_url'], '')
+    def test_create_video_access_error(self) -> None:
+        class MockResponse:
+            def __init__(self) -> None:
+                self.status_code = 200
 
-    def test_create_video_request(self) -> None:
-        with mock.patch('requests.post'):
-            result = self.client_get("/json/calls/create")
-            self.assert_json_success(result)
+            def json(self) -> Dict[str, str]:
+                return {"access_token": "token"}
+
+            def raise_for_status(self) -> None:
+                return None
+
+        with mock.patch('requests.post', return_value=MockResponse()):
+            result = self.client_get("/json/calls/complete_zoom_user?code=code")
+            self.assert_json_error(result, "Invalid Zoom access token")
+
+    def test_create_video_request_error(self) -> None:
+        class MockResponse:
+            def __init__(self) -> None:
+                self.status_code = 200
+
+            def json(self) -> Dict[str, str]:
+                return {"access_token": "token", "join_url": "example.com"}
+
+            def raise_for_status(self) -> None:
+                return None
+
+        class MockResponseFail:
+            def __init__(self) -> None:
+                self.status_code = 200
+
+            def raise_for_status(self) -> None:
+                raise Exception()
+
+        class MockResponseGet:
+            def __init__(self) -> None:
+                self.status_code = 200
+
+            def json(self) -> Dict[str, str]:
+                return {"id": "id"}
+
+            def raise_for_status(self) -> None:
+                return None
```

Comment:
You might be able to do better using the `httpretty` library rather than creating a bunch of mock classes; that's more designed for this sort of purpose.